File: holly_ppo_biblioteca_sin_input/conexion.py
import mysql.connector
class Conectar:
    def __init__(self):
        self.conexion = None

    def conexion_bdd(self):
        try:    
            self.conexion = mysql.connector.connect(
                host='localhost',
                user='root',
                password='', #yo tengo que ponerlo en blanco porque al poner la contraseña curso que tenemos todos, salta un error, no me digas por qué.
                database='biblioteca'
            )
            print("Conexión exitosa.")

        except mysql.connector.Error as e:
            print("Error al conectar a la base de datos: ", e)

    def close_connection(self):
        if self.conexion:
            self.conexion.close()
            print("Conexión cerrada.")
    
        

 

    

File: apuntes_php_ppo/conexion.php
<?php

class ConexionBiblioteca {

    private $conexion;

    public function conectar() {
        $this->conexion = new mysqli('localhost', 'root', 'DAM1', 'biblioteca');

        if ($this->conexion->connect_error) {
            die("Error en conexión: " . $this->conexion->connect_error);
        }

        echo "Conexión exitosa.";
    }

    public function cierra_conexion() {
        if ($this->conexion) {
            $this->conexion->close();
            echo "Conexión cerrada.";
        }
    }
}